### PR TITLE
Generate Random Meal/By ID Use Case

### PR DESCRIPTION
**Pull request by merrickliu888**
### Changes
- Created a use case for generating meals by an id, for the purpose of the generate random meal use case. This interactor serves as the backend of the generate random meal use case, and offers extensibility for other use cases that may require generating meals by a specific id.
- I also made some small refactors to the `GenereateMeal` interactor where I just added some spacing. @jackkfan0305 
- Also @samaslanowicz your view will be calling this use case in your view too.

### Testing
In terms of test, I tested the controller for `GenerateMealByID` and the `.execute()` of the interactor. Here is my coverage:
**Controller** - Made sure that the random meal id fetched is valid.
image
**Interactor** - Made sure that the correct meal is fetched and the json is correctly parsed into the output data object.
image

### CODE REVIEW

File: test/interface_adapter/generate_random_meal/GenerateRandomMealControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                        String selectedRecipe = inputData.getMealID();
+
+                        for (String recipe : recipes) {
+                            if (recipe.equals(selectedRecipe)) {
+                                inRecipeList = true;
+                                break;
+                            }
+                        }
+
+                        assertTrue(inRecipeList);
```
Comment (jackkfan0305) [inline test/interface_adapter/generate_random_meal/GenerateRandomMealControllerTest.java:30]: Good test to see if a random selected recipe is in the `randomRecipesList` arraylist. I'm not sure if this is necessary, but do you think the test would be more comprehensive if we test if, for example, 100 randomly selected recipes belongs to the `randomRecipesList` arraylist? Due to the fact that the recipe index is randomly generated, testing for one random recipe may be due to luck.

Comment (merrickliu888) [reply test/interface_adapter/generate_random_meal/GenerateRandomMealControllerTest.java:30]: Yes. Makes sense I will do that.

**Review by samaslanowicz — COMMENTED** [review]
looks good to me

**Review by samaslanowicz — APPROVED** [review]
looks good

**Pull request merged**